hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.test.tsx

```diff
@@ -108,4 +118,91 @@ describe("Inventory Page", () => {
         expect(queryByTestId("inventoryCountDivider")).not.toBeInTheDocument();
         expect(getByText(/no items found/i)).toBeInTheDocument();
     });
+
+    it("Loads more hardware", async () => {
+        // Mock hardware api response with a limit of mockHardware.length - 2
+        const limit = mockHardware.length - 2;
+        const nextURL = `http://localhost:8000${hardwareUri}?offset=${limit}`;
+        const hardwareApiResponse = makeMockApiListResponse(
+            mockHardware.slice(0, limit),
+            nextURL,
+            null,
+            mockHardware.length
+        );
+        const hardwareNextApiResponse = makeMockApiListResponse(
+            mockHardware.slice(limit),
+            null,
+            hardwareUri,
+            mockHardware.length
+        );
+        const categoryApiResponse = makeMockApiListResponse(mockCategories);
+
+        when(mockedGet)
+            .calledWith(hardwareUri, {})
+            .mockResolvedValue(hardwareApiResponse);
+        when(mockStripHostnameReturnFilters)
```

Comment:
Going off the comment above, I don't think you should be mocking this. You know what data is being passed in to this function, so you should know what it's output is. There aren't any side-effects either, so it's safe to not mock I think?

Even better, you can do a `expect(mockedGet).toHaveBeenCalledWith(/* whatever you expect stripHostnameReturnFiletrs to return */)`, that way you have a test for that function.

Reply:
ah okay, going off my reply to your comment above lol I'll try this to see if it solved my problem

Reply:
nvm lol it doesn't work :( I need the when()